src/zarr/strategies.py

```diff
@@ -143,3 +145,19 @@ def basic_indices(draw: st.DrawFn, *, shape: tuple[int], **kwargs):  # type: ign
             )
         )
     )
+
+
+def key_ranges(keys: SearchStrategy | None = None) -> st.lists[:]:
```

Comment:
```suggestion
def key_ranges(keys: SearchStrategy | None = None) -> SearchStrategy[list]:
```
https://hypothesis.readthedocs.io/en/latest/details.html#writing-downstream-type-hints

You can specialize the inner `list` type a lot more but this is the syntax.